edtf/parser/parser_classes.py

```diff
@@ -276,6 +277,26 @@ def __init__(
             int(significant_digits) if significant_digits else None
         )
 
+    def set_year(self, y: int):
+        if y is None:
+            raise AttributeError("Year must not be None")
+        self._year = y
+
+    def get_year(self) -> int:
+        return self._year
+
+    year = property(get_year, set_year)
+
+    def set_month(self, m: Optional[int]):
+        self._month = m
+        if m is None:
+            self.day = None
+
+    def get_month(self) -> Optional[int]:
+        return self._month
+
+    month = property(get_month, set_month)
```

Comment:
These methods duplicate the ones above. I'm not sure if that's just a weird merge thing though?